src/model/UserphraseModel.cpp

```diff
@@ -218,3 +220,13 @@ const Userphrase *UserphraseModel::getUserphrase(const QModelIndex& idx)
 {
     return &userphrase_[idx.row()];
 }
+
+void UserphraseModel::undo()
+{
+    if(userphrase_tmp_.size() > 0) {
```

Comment:
use `empty()` and add space between `if` and parenthesis:
`if (!userphrase_tmp_.empty()) {`